plugins/chat/assets/javascripts/discourse/components/chat-composer.js

```diff
@@ -1,172 +1,249 @@
 import { setupHashtagAutocomplete } from "discourse/lib/hashtag-autocomplete";
-import {
-  chatComposerButtons,
-  chatComposerButtonsDependentKeys,
-} from "discourse/plugins/chat/discourse/lib/chat-composer-buttons";
-
-const THROTTLE_MS = 150;
-
-export default Component.extend(TextareaTextManipulation, {
-  chatChannel: null,
-  chat: service(),
-  classNames: ["chat-composer-container"],
-  classNameBindings: ["emojiPickerVisible:with-emoji-picker"],
-  chatEmojiReactionStore: service("chat-emoji-reaction-store"),
-  chatEmojiPickerManager: service("chat-emoji-picker-manager"),
-  chatStateManager: service("chat-state-manager"),
-  timer: null,
-  value: "",
-  inProgressUploads: null,
-  composerEventPrefix: "chat",
-  composerFocusSelector: ".chat-composer-input",
-  canAttachUploads: reads("siteSettings.chat_allow_uploads"),
-  isNetworkUnreliable: reads("chat.isNetworkUnreliable"),
-  typingMention: false,
-  chatComposerWarningsTracker: service(),
-
-  @discourseComputed(...chatComposerButtonsDependentKeys())
-  inlineButtons() {
-    return chatComposerButtons(this, "inline", this.context);
-  },
-
-  @discourseComputed(...chatComposerButtonsDependentKeys())
-  dropdownButtons() {
-    return chatComposerButtons(this, "dropdown", this.context);
-  },
-
-  @discourseComputed("chatEmojiPickerManager.{opened,context}")
-  emojiPickerVisible(picker) {
-    return picker.opened && picker.context === "chat-composer";
-  },
-
-  @discourseComputed("chatStateManager.isFullPageActive")
-  fileUploadElementId(fullPage) {
-    return fullPage ? "chat-full-page-uploader" : "chat-widget-uploader";
-  },
-
-  init() {
-    this._super(...arguments);
-
-    this.appEvents.on(
-      "upload-mixin:chat-composer-uploader:in-progress-uploads",
-      this,
-      "_inProgressUploadsChanged"
+import { isPresent } from "@ember/utils";
+
+export default class ChatComposer extends Component {
+  @service capabilities;
+  @service site;
+  @service siteSettings;
+  @service chat;
+  @service chatComposerPresenceManager;
+  @service chatComposerWarningsTracker;
+  @service appEvents;
+  @service chatEmojiReactionStore;
+  @service chatEmojiPickerManager;
+
+  @tracked isFocused = false;
+
+  get pane() {
+    return this.args.paneService;
+  }
+
+  get composer() {
+    return this.args.composerService;
+  }
+
+  get inlineButtons() {
+    return chatComposerButtons(this, "inline", this.args.context);
+  }
+
+  get dropdownButtons() {
+    return chatComposerButtons(this, "dropdown", this.args.context);
+  }
+
+  get fileUploadElementId() {
+    return this.args.context + "-file-uploader";
+  }
+
+  get canAttachUploads() {
+    return (
+      this.siteSettings.chat_allow_uploads &&
+      isPresent(this.args.uploadDropZone)
     );
+  }
 
-    this.setProperties({
-      inProgressUploads: [],
-      _uploads: [],
-    });
+  @action
+  setupAutocomplete(textarea) {
+    const $textarea = $(textarea);
```

Comment:
I wonder if we should just set this ref inside `constructor` for `TextareaInteractor`, then you could have `this._$textarea`? Maybe would make it easier to kill later for jQuery

Reply:
Ok fair enough :+1: